Adds Combat mode, rclick functionality

Tsar-Salat opened the pull request:
## About The Pull Request

This adds right-click functionality through the addition of the combat mode system.

Changes are as listed:

### Major Codebase changes

- Disarm and Grab Intents have been removed.
- Diametric intents (harm/help) are not combat mode. Toggled by F or 4.
- Context menu now appears on shift+rclick (like examine)
- Disarm functionality has now been moved to rclick, whether CM is active or not
- Grabbing has been centralized in its hotkey ctrl-click

Functional Item changes
- Ninja Dash now activates via right-click
- M90 grenade launcher activates through right click rather than toggle
- Health analyzer now scans health with left click, chemicals with right click
- Jumpsuits change suit sensors through right-click
- Airlocks are now welded through rightclick
- Reagent splashing is done through right click
- RCDs deconstruct through rclick
- Syringes now lclick to inject, rclick to draw 
- washingmachines now use rclick to start
- stack splitting now uses rclick
- hookshotguns use rclick instead of toggle
- deconstructing windows, tables, chairs is done through rclick with appropriate tools
- secure lockers are now (un)locked through rclick
- gas canisters are repaired on lclick and deconstructed on rclick
- right clicking a reagent container will now set it to its previous transfer amount
- Screwdrivers can now be put in autolathes through lclick, maintenance through rclick
- girders can be built by rclicking a turf with a sheet of iron (allowing building multiple at the same time)
- Most previously rotatable structures or machines can now be rotated through mouseclick. Left for Counterclockwise, right for clockwise.
- closet wrench actions moved to rclick
- chemistry machines use rclick to add/remove their beakers
- Kinetic Crusher projectiles are now fired on rclick
- ID Painter now lets your remove IDs/PDA through rclick
- PKA modkits can now be removed through a radial on rclick
- storage item access has been moved from alt+click to rclick
- stunbatons now stunshove through rclick, instead of on disarm intent

<details>
<summary>Ports:</summary>

(not rclick but rclick needed)
- https://github.com/tgstation/tgstation/pull/44084
- https://github.com/tgstation/tgstation/pull/45132
- https://github.com/tgstation/tgstation/pull/45502
- https://github.com/tgstation/tgstation/pull/46497
- https://github.com/tgstation/tgstation/pull/46568
- https://github.com/tgstation/tgstation/pull/46570
- https://github.com/tgstation/tgstation/pull/47685 (so all mobs support it)
- https://github.com/tgstation/tgstation/pull/47955
- https://github.com/tgstation/tgstation/pull/48766
- https://github.com/tgstation/tgstation/pull/52541
- https://github.com/tgstation/tgstation/pull/52699
- https://github.com/tgstation/tgstation/pull/54665
- https://github.com/tgstation/tgstation/pull/54747
- https://github.com/tgstation/tgstation/pull/54769 (test combat stuff)
- https://github.com/tgstation/tgstation/pull/54842
- https://github.com/tgstation/tgstation/pull/55305
- https://github.com/tgstation/tgstation/pull/55583
- https://github.com/tgstation/tgstation/pull/56520
- https://github.com/tgstation/tgstation/pull/57131
- https://github.com/tgstation/tgstation/pull/57167
- https://github.com/tgstation/tgstation/pull/68552
- https://github.com/tgstation/tgstation/pull/71044 (only refactors for rclick)
- https://github.com/tgstation/tgstation/pull/79047




Rclick:
- https://github.com/tgstation/tgstation/pull/56601
- https://github.com/tgstation/tgstation/pull/56636
- https://github.com/tgstation/tgstation/pull/56667
- https://github.com/tgstation/tgstation/pull/56699
- https://github.com/tgstation/tgstation/pull/56700
- https://github.com/tgstation/tgstation/pull/56779
- https://github.com/tgstation/tgstation/pull/56781
- https://github.com/tgstation/tgstation/pull/56811
- https://github.com/tgstation/tgstation/pull/56813
- https://github.com/tgstation/tgstation/pull/56816
- https://github.com/tgstation/tgstation/pull/56829
- https://github.com/tgstation/tgstation/pull/56846
- https://github.com/tgstation/tgstation/pull/56864
- https://github.com/tgstation/tgstation/pull/56865
- https://github.com/tgstation/tgstation/pull/57214
- https://github.com/tgstation/tgstation/pull/57617
- https://github.com/tgstation/tgstation/pull/57788
- https://github.com/tgstation/tgstation/pull/57837
- https://github.com/tgstation/tgstation/pull/57841
- https://github.com/tgstation/tgstation/pull/57881
- https://github.com/tgstation/tgstation/pull/58124
- https://github.com/tgstation/tgstation/pull/58270
- https://github.com/tgstation/tgstation/pull/58523
- https://github.com/tgstation/tgstation/pull/58774
- https://github.com/tgstation/tgstation/pull/58802
- https://github.com/tgstation/tgstation/pull/58830
- https://github.com/tgstation/tgstation/pull/59508
- https://github.com/tgstation/tgstation/pull/59656
- https://github.com/tgstation/tgstation/pull/59814
- https://github.com/tgstation/tgstation/pull/60653
- https://github.com/tgstation/tgstation/pull/60990
- https://github.com/tgstation/tgstation/pull/61275
- https://github.com/tgstation/tgstation/pull/61795
- https://github.com/tgstation/tgstation/pull/62949
- https://github.com/tgstation/tgstation/pull/63038
- https://github.com/tgstation/tgstation/pull/64441
- https://github.com/tgstation/tgstation/pull/64482
- https://github.com/tgstation/tgstation/pull/64724
- https://github.com/tgstation/tgstation/pull/64939
- https://github.com/tgstation/tgstation/pull/64964
- https://github.com/tgstation/tgstation/pull/65353
- https://github.com/tgstation/tgstation/pull/65978
- https://github.com/tgstation/tgstation/pull/66038
- https://github.com/tgstation/tgstation/pull/66928
- https://github.com/tgstation/tgstation/pull/71603
- https://github.com/tgstation/tgstation/pull/72463
- https://github.com/tgstation/tgstation/pull/72959
- https://github.com/tgstation/tgstation/pull/76057

</details>


## Why It's Good For The Game

Intents are by-and-large not a very well written system. The "grab" intent, for instance, has had a hotkey for years on ctrl-click.

The large goal of this PR is to address two fundamental issues:

1. The current intent check system, which almost entirely relies on if a character is or is not on help or harm mode, to determine that players intention.
2. The clunkiness of actual systems that DO fully utilize the intent system

For the first issue, nearly all of our intent checks in the codebase are just diametric checks for if a player is in harm or help mode. Grab is only checked when checking grab_state, which is attached to attack_hand, otherwise you can basically forget about the intent and just use the ctrl hotkey. 
Disarm is in an even worse boat, its basically Harm Jr. in that players in combat will primarily be on harm while in combat and be spam-switching between the two *only* for the purpose of trying to stun them on a wall, table, or mob.

For the second issue, the most notable use of intents is martial arts. This is *not* to the boon of martial arts either, as the combination of moves that you have to do *mid-combat* is not very well done. 

Allowing players to free up their mouse instead of the specific intents gives coders a lot more room and buttons to work with, and allow simplification for needlessly complex tasks like martial arts or construction.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

rclick welding

https://github.com/user-attachments/assets/a57beea5-e2fc-4a6f-ada5-8157d5f468ca

lclick/rclick syringes

https://github.com/user-attachments/assets/24dbf3ec-9cf3-4f29-8442-86063871daf1

grab states

https://github.com/user-attachments/assets/f967f636-702e-432a-bca3-d63fcad7b1ce

RCD rclick deconstruction

https://github.com/user-attachments/assets/2ff9beba-18e2-4ee7-80cb-04fd66c104e4

generic structure deconstruction

https://github.com/user-attachments/assets/98534cc5-569b-46ef-9112-2e16a07425ea

CW/CCW mouse rotation

https://github.com/user-attachments/assets/b8cfabca-352a-4d9a-a08d-55bdf6007236

rclick girder construction

https://github.com/user-attachments/assets/b8b80e36-f3d7-4c2a-b978-64f258750114



</details>

## Changelog
:cl: rkz, Qustinnus, Mothblocks, TJatPBnJ, IndieanaJones, KathyRyals, Arkatos, norill, Fikou, dragomagol, SgtHunk, JJRCop, ninjanomnom, cacogen, Timberpoes, LemonInTheDark, Ghommie, Sealed101, Ghilker, TheBonded, Timothymtorres, CameronWoof, ArcaneDefence, BurgerLUA, OneAsianTortoise, vincentiusvin, itseasytosee, AnturK, GuillaumePrata, Timberpoes, Ghommie, zxaber, Gpeckman, haukeschaumann, tralezab, Jack LeCroy, MrMelbert, ShizCalev, Actioninja
add: Adds combat mode
add: Adds martial art combo meters and TGUI guides
refactor: refactors CanUseTopic and dexterity
refactor: refactored syringe code to use flags
tweak: clicking on the turf of a airlock while combat mode is off will now close the airlock. No more pixel hunting!
balance: buffs to hook shotgun in addition to rclick functionality
code: renames RemoveComponent, component destroy will no longer runtime when parent is null
refactor: refactored simple rotation component
fix: fixed abandoned crates to not try to kill you if you perform actions on them AFTER they are unlocked
code: introduced variable weapon attack speed. No weapon utilizes this yet but admins can now varedit items to attack faster/slower.
code: unit tests reagent container transfer amounts
code: unit tests attacking people

balance: Using Jaunts will conceal your runechat for their duration.
code: passed IS_HERETIC through the proc, rather than checking everytime you used a heretic spell. Less runtimes.

fix: cutting wires while a TGUI menu is open will no longer shock you and bypass shock protection
tweak: cleaned up airlock prying code. Nightmare blades are now crowbars and fulfill tool_acts, they still can break stuff by being on combat mode though. Rclick with a one to pry regardless of intent.

refactor: refactors stun batons
tweak: abductor batons are children of stun batons
/:cl:

Tsar-Salat committed:
butchering fix

Tsar-Salat committed:
INITIAL DONE!

Tsar-Salat committed:
alt

Tsar-Salat committed:
renaming, add suit sensor interaction

Tsar-Salat committed:
preattacksecondary

https://github.com/tgstation/tgstation/pull/56700

Tsar-Salat committed:
attackhandsecondary

https://github.com/tgstation/tgstation/pull/56781

Tsar-Salat committed:
fix pre_attack

https://github.com/tgstation/tgstation/pull/56798

Tsar-Salat committed:
lazy access modifiers and ninja dash

https://github.com/tgstation/tgstation/pull/56636 https://github.com/tgstation/tgstation/pull/56846

Tsar-Salat committed:
secondary tool acts

https://github.com/tgstation/tgstation/pull/58830

Tsar-Salat committed:
drones without drones (big toolact refactor)

https://github.com/tgstation/tgstation/pull/58802 https://github.com/tgstation/tgstation/pull/66928

Tsar-Salat committed:
right click reagent containers

https://github.com/tgstation/tgstation/pull/59508

Tsar-Salat committed:
redo rightclick priority

https://github.com/tgstation/tgstation/pull/59656

Tsar-Salat committed:
final touches on attack chain

Tsar-Salat committed:
airlock welding rclick

https://github.com/tgstation/tgstation/pull/56811

Tsar-Salat committed:
reagent container unit test

https://github.com/tgstation/tgstation/pull/76057

Tsar-Salat committed:
rightclick bottles

https://github.com/tgstation/tgstation/pull/56813

Tsar-Salat committed:
ranged attack secondary

https://github.com/tgstation/tgstation/pull/57214 https://github.com/tgstation/tgstation/pull/57841 https://github.com/tgstation/tgstation/pull/56667 https://github.com/tgstation/tgstation/pull/56829

Tsar-Salat committed:
rcd rclick

https://github.com/tgstation/tgstation/pull/56864

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

Tsar-Salat committed:
medical rclick

https://github.com/tgstation/tgstation/pull/56779

Tsar-Salat committed:
syringe rclick

https://github.com/tgstation/tgstation/pull/56816 https://github.com/tgstation/tgstation/pull/56865

Tsar-Salat committed:
crowbar

Tsar-Salat committed:
ctrlclick

https://github.com/tgstation/tgstation/pull/57617

Tsar-Salat committed:
washing machine

https://github.com/tgstation/tgstation/pull/57788

Tsar-Salat committed:
stack splitting rclick

https://github.com/tgstation/tgstation/pull/57837

Tsar-Salat committed:
fix ai attacks

https://github.com/tgstation/tgstation/pull/57881

Tsar-Salat committed:
floortile airlock interaction

https://github.com/tgstation/tgstation/pull/58124

Tsar-Salat committed:
fix borg hug

Tsar-Salat committed:
hook shotgun rclick(buff)

https://github.com/tgstation/tgstation/pull/58270

Tsar-Salat committed:
rpd rclick

https://github.com/tgstation/tgstation/pull/58309

Tsar-Salat committed:
Revert "rpd rclick"

This reverts commit 4cac0d2969b80ce9b1d88f0d3ef68bc6be56fe0c.

Tsar-Salat committed:
window rclick

https://github.com/tgstation/tgstation/pull/58523

Tsar-Salat committed:
closet lock rclick

https://github.com/tgstation/tgstation/pull/58774

Tsar-Salat committed:
microwave and circuit rclick

Tsar-Salat committed:
fix rclick storage stuff

Tsar-Salat committed:
I swore I did this already lmao

Tsar-Salat committed:
fix locker secndary. Again.

https://github.com/tgstation/tgstation/pull/59814

Tsar-Salat committed:
fixes monkeycode

https://github.com/tgstation/tgstation/pull/60990

Tsar-Salat committed:
autolathe screwdriver rclick

https://github.com/tgstation/tgstation/pull/61275

Tsar-Salat committed:
space heater rclick

Tsar-Salat committed:
girder construction rclick

https://github.com/tgstation/tgstation/pull/61795

Tsar-Salat committed:
rclick wrench deconstruct

https://github.com/tgstation/tgstation/pull/63038

Tsar-Salat committed:
simplerotation lmb rmb

https://github.com/tgstation/tgstation/pull/60653 https://github.com/tgstation/tgstation/pull/72463 https://github.com/tgstation/tgstation/pull/64441

Tsar-Salat committed:
wrench closets

https://github.com/tgstation/tgstation/pull/64724

Tsar-Salat committed:
kinetic crusher rclick

https://github.com/tgstation/tgstation/pull/64964

Tsar-Salat committed:
fixes abandoned crates

https://github.com/tgstation/tgstation/pull/62949 https://github.com/tgstation/tgstation/pull/64482 https://github.com/tgstation/tgstation/pull/65353

Tsar-Salat committed:
chem machine rclick

https://github.com/tgstation/tgstation/pull/65978 https://github.com/tgstation/tgstation/pull/64939

Tsar-Salat committed:
ladders lmb rmb

https://github.com/tgstation/tgstation/pull/66038

Tsar-Salat committed:
cleanup clickon.

Tsar-Salat committed:
fixes

Tsar-Salat committed:
invert handcuff?

Tsar-Salat committed:
try this instead

Tsar-Salat committed:
variable weapon attack speed

https://github.com/tgstation/tgstation/pull/72959

Tsar-Salat committed:
guh

Tsar-Salat committed:
TGUI THIS SHIRT

Tsar-Salat committed:
oops

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

PowerfulBacon reviewed (commented):
I am running this locally and the hotkeys F/4 don't show up in the preferences menu and don't work. Do the hotkeys for this action actually exist?

The context menu is also enabled by default, and to turn it off you have to first press the shift key.

Holding shift also doesn't show the examine magnifying glass like other servers have.

Tsar-Salat committed:
misc fixes

Tsar-Salat committed:
add keybind

Tsar-Salat committed:
add custom cursors + shiftclick magnifier

https://github.com/tgstation/tgstation/pull/54083

Tsar-Salat committed:
fix right click not updated

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

Tsar-Salat committed:
fix modkit

https://github.com/tgstation/tgstation/pull/54842

Tsar-Salat committed:
rclick modkits

Tsar-Salat committed:
minebots and cyborgs are still a little effed, this should work

https://github.com/tgstation/tgstation/pull/79047, https://github.com/tgstation/tgstation/pull/55305

Tsar-Salat committed:
fix

Tsar-Salat committed:
wiremod adjustments

Tsar-Salat committed:
combat unit test

https://github.com/tgstation/tgstation/pull/54769

Tsar-Salat committed:
drop items

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

Tsar-Salat committed:
herrm

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

Tsar-Salat committed:
CQC text fix

Tsar-Salat committed:
grab fixes (for humans)

https://github.com/tgstation/tgstation/pull/49544 https://github.com/tgstation/tgstation/pull/51427 https://github.com/tgstation/tgstation/pull/54294
https://github.com/tgstation/tgstation/pull/64642

Tsar-Salat committed:
monkeys dont hate you anymore

https://github.com/tgstation/tgstation/pull/62685

Tsar-Salat committed:
moving to grip code

Tsar-Salat committed:
MARTIAL ARTS CODE IS ASSS

https://github.com/tgstation/tgstation/pull/45502
https://github.com/tgstation/tgstation/pull/46497
https://github.com/tgstation/tgstation/pull/55583
https://github.com/tgstation/tgstation/pull/56397

Tsar-Salat committed:
ctrl and more martials

https://github.com/tgstation/tgstation/pull/57617
https://github.com/tgstation/tgstation/pull/57131

Tsar-Salat committed:
atomized CanUseTopic() refactor

Tsar-Salat committed:
cqc and sleeping carp

https://github.com/tgstation/tgstation/pull/44084
https://github.com/tgstation/tgstation/pull/45132

Tsar-Salat committed:
martial cleanup

https://github.com/tgstation/tgstation/pull/45502
https://github.com/tgstation/tgstation/pull/46497

Tsar-Salat committed:
plasmafist

https://github.com/tgstation/tgstation/pull/47955

Tsar-Salat committed:
Monkey Martial Arts

Tsar-Salat committed:
combo meter

https://github.com/tgstation/tgstation/pull/56520
https://github.com/tgstation/tgstation/pull/57167
https://github.com/tgstation/tgstation/pull/57131

Tsar-Salat committed:
chisel it into stone

https://github.com/tgstation/tgstation/pull/57131

Tsar-Salat committed:
fixes generic huds

Tsar-Salat committed:
ctrlclick fix

https://github.com/tgstation/tgstation/pull/57617

Tsar-Salat committed:
pacifism martial arts

https://github.com/tgstation/tgstation/pull/72183

Tsar-Salat committed:
cqc knockouts

https://github.com/tgstation/tgstation/pull/68552

Tsar-Salat committed:
make the combo menu show up for monkeys

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodeatomization-canusetopic

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into martial-artspain

Tsar-Salat committed:
.

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into martial-artspain

Tsar-Salat committed:
remove conflict

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodeatomization-canusetopic

Tsar-Salat committed:
.

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into martial-artspain

Tsar-Salat committed:
Merge branch 'martial-artspain' into combatmodetake5

Tsar-Salat committed:
.

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodeatomization-canusetopic

Tsar-Salat committed:
Merge branch 'combatmodeatomization-canusetopic' into combatmodetake5

Tsar-Salat committed:
fix attack chain bug

https://github.com/tgstation/tgstation/pull/46842

Tsar-Salat committed:
grrrr

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

Tsar-Salat committed:
trigger flags (not my issue)

Tsar-Salat committed:
fix combat mode toggle pref, firelocks

Tsar-Salat committed:
fix cyborg huds & throat slicing

Tsar-Salat committed:
fix stack splitting and health analyzer modes

Tsar-Salat committed:
move grabbing to carbon level

Tsar-Salat committed:
martial arts menu

Tsar-Salat committed:
fix martial arts formatting

Tsar-Salat committed:
fix

Tsar-Salat committed:
move firedoor welding to rclick

Tsar-Salat committed:
fix cyborg action

Tsar-Salat committed:
fix martials, again

Tsar-Salat committed:
airlock icon update hotfix

Tsar-Salat committed:
open na noor... open na noor!!!11!

Tsar-Salat committed:
fix camera actions

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

Tsar-Salat committed:
fix piggybacking

Tsar-Salat committed:
fixes the combo once and for all

Tsar-Salat committed:
tribal claw combos

Tsar-Salat committed:
we all pass list modifiers. Always.

Tsar-Salat committed:
firealarm cleanup

Tsar-Salat committed:
fix supplypod launcher

PowerfulBacon commented on interface/skin.dmf:
```diff
@@ -1,4 +1,12 @@
 macro "default"
+	elem ".winset :map.right-click=false"
+		name = "SHIFT+Shift"
```
What is this macro for and why is the winset command its name?

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

PowerfulBacon committed:
Fixes ghosts needing to shift click

PowerfulBacon committed:
Merge branch 'master' into combatmodetake5

Tsar-Salat committed:
rclick food storage

Tsar-Salat committed:
fix some mergeskewed huds

Tsar-Salat committed:
fix chem dispenser wrench act

Tsar-Salat committed:
fix heretic spells, adds support to rclick spells, makes mansus use rclick

https://github.com/tgstation/tgstation/pull/71044

Tsar-Salat committed:
fix bad attack chain, genericize crowbar-likes into actual crowbars, use door pryer trait as check

https://github.com/tgstation/tgstation/pull/46570, https://github.com/tgstation/tgstation/pull/46568, https://github.com/tgstation/tgstation/pull/52699,
https://github.com/tgstation/tgstation/pull/52541,
https://github.com/tgstation/tgstation/pull/52699,

Tsar-Salat committed:
I like neckslicing

Tsar-Salat committed:
hidden runechat fix

Tsar-Salat committed:
work on zombies

Tsar-Salat committed:
hm

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

Tsar-Salat committed:
macro

Tsar-Salat committed:
aggrograb

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5

Tsar-Salat committed:
stunbaton cleanup, move stunshove to rclick

https://github.com/tgstation/tgstation/pull/48766

Tsar-Salat committed:
fix your goddamn params, proc_holders!

https://github.com/tgstation/tgstation/pull/57345

Tsar-Salat committed:
mapedit

Tsar-Salat committed:
tool_act fixes

https://github.com/tgstation/tgstation/pull/68278,
https://github.com/tgstation/tgstation/pull/65403

PowerfulBacon reviewed (approved):
Can I merge this and get continued support in a new PR?

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into combatmodetake5
